src/modules/sway/bar.cpp

```diff
@@ -61,38 +66,108 @@ void BarIpcClient::onInitialConfig(const struct Ipc::ipc_response& res) {
 void BarIpcClient::onIpcEvent(const struct Ipc::ipc_response& res) {
   try {
     auto payload = parser_.parse(res.payload);
-    if (auto id = payload["id"]; id.isString() && id.asString() != bar_.bar_id) {
-      spdlog::trace("swaybar ipc: ignore event for {}", id.asString());
-      return;
-    }
-    if (payload.isMember("visible_by_modifier")) {
-      // visibility change for hidden bar
-      signal_visible_(payload["visible_by_modifier"].asBool());
-    } else {
-      // configuration update
-      auto config = parseConfig(payload);
-      signal_config_(std::move(config));
+    switch (res.type) {
+      case static_cast<uint32_t>(IPC_EVENT_WORKSPACE):
+        if (payload.isMember("change")) {
+          // only check and send signal if the workspace update reason was because of a urgent
+          // change
+          if (payload["change"] == "urgent") {
+            // Retrieve status for all workspace in order to check if urgency is still active for at
+            // least one workspace
+            ipc_.sendCmd(IPC_GET_WORKSPACES);
+          }
+          last_press_modifier_ = false;
+        }
+        break;
+      case static_cast<uint32_t>(IPC_EVENT_MODE):
+        if (payload.isMember("change")) {
+          signal_mode_(payload["change"] != "default");
+          last_press_modifier_ = false;
+        }
+        break;
+      case static_cast<uint32_t>(IPC_EVENT_BINDING):
+        last_press_modifier_ = false;
+        break;
+      case static_cast<uint32_t>(IPC_EVENT_BAR_STATE_UPDATE):
+      case static_cast<uint32_t>(IPC_EVENT_BARCONFIG_UPDATE):
+        if (auto id = payload["id"]; id.isString() && id.asString() != bar_.bar_id) {
+          spdlog::trace("swaybar ipc: ignore event for {}", id.asString());
+          return;
+        }
```

Comment:
We should have this condition for all type, no?

Reply:
No, only these 2 events can be sent to a specific instance of a bar and may have bar_id.